File: src/BarberShop.Domain/Appointment.cs
using BarberShop.Communication.Enums.Appointment;

namespace BarberShop.Domain
{
    public class Appointment : BaseEntity
    {
        public long UserId { get; set; }
        public long? BarberId { get; set; }   // nullable for backward compat
        public long? ServiceId { get; set; }  // nullable for backward compat
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public EAppointmentStatus Status { get; set; } = EAppointmentStatus.WaitingPayment;

        public User User { get; set; } = null!;
        public Barber? Barber { get; set; }
        public Service? Service { get; set; }
        public Payment? Payment { get; set; }
    }
}

File: frontend/src/types/appointment.ts
export type AppointmentStatus = 0 | 1 | 2 // WaitingPayment | Paid | Cancelled

export interface Appointment {
  id: number
  userId: number
  startTime: string
  endTime: string
  status: AppointmentStatus
  creationDate: string
  updateDate?: string
}

export interface CreateAppointmentRequest {
  userId: number
  date: string       // "YYYY-MM-DD"
  startTime: string  // "HH:mm"
  serviceId: number
  barberId: number
}

export interface CancelAppointmentRequest {
  appointmentId: number
  userId: number
}
